### PR TITLE
#170764582 Logout functionality

### PR DESCRIPTION
**Pull request by shaazk**
What does this PR do?
- Enable a user to logout from the app

Description of Task to be completed?
- Add logout endpoint test
- Add the user token to a blacklist
- Add user logout tests

How should this be manually tested?
- git clone this repo and checkout to ft-logout-functionality-170764582
- run npm i to install dependencies
- install a redis and configure it guide
- update the .env with REDIS_URL=redis://127.0.0.1:6379
- run the app with npm run dev
- test the endpoint at 127.0.0.1:5000/api/auth/logout

Any background context you want to provide?
N/A

What are the relevant pivotal tracker stories?
[#170764582](https://www.pivotaltracker.com/story/show/170764582)

What are the relevant Github Issues?
N/A
Screenshots (if appropriate)

Screenshot from 2020-01-29 13-46-00
Screenshot from 2020-01-29 13-46-44
Screenshot from 2020-01-29 13-47-34
Screenshot from 2020-01-29 13-46-29

### CODE REVIEW

**Review by niyodusengaclement — COMMENTED** [review]
LGTM

File: src/repositories/userRepository.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  }
+
+  /**
+     *
+     * @param {string} email
+     * @returns {obj} record is object if email found or null if not
+     */
+  async findByEmail(email) {
+    try {
+      const record = await this.db.findOne(
```
Comment (niyodusengaclement) [inline src/repositories/userRepository.js]: Remember to enable eslint extension in your code editor

Comment (shaazk) [reply src/repositories/userRepository.js]: it is enabled

File: src/database/seeders/20200121122716-create-user.js [hunk 2]
```diff
@@ -8,5 +8,5 @@ export default {
     role: 'super-administrator'
   }], {}),
 
-  down: (queryInterface) => queryInterface.bulkDelete('User', null, {})
+  down: (queryInterface) => queryInterface.bulkDelete('Users', null, {})
```
Comment (ajimae) [inline src/database/seeders/20200121122716-create-user.js]: I think the rule you guys configured for houndCI doesn't allow something like this:
`down: (queryInterface) => queryInterface.bulkDelete('User', null, {})` 

So try this instead
`down: queryInterface => queryInterface.bulkDelete('User', null, {})`

File: src/database/seeders/20200121122716-create-user.js [hunk 3]
```diff
@@ -1,5 +1,5 @@
 export default {
-  up: (queryInterface) => queryInterface.bulkInsert('User', [{
+  up: (queryInterface) => queryInterface.bulkInsert('Users', [{
```
Comment (ajimae) [inline src/database/seeders/20200121122716-create-user.js]: Same this here, you either remove the parentheses around `queryInterface` or add curly braces around the body like this:

```
down: (queryInterface) => {
   queryInterface.bulkDelete('User', null, {})
}
```

This should fix it

**Review by Elvis-rugamba — COMMENTED** [review]
remove hound errors

**Review by ajimae — COMMENTED** [review]
LGTM!

**Pull request merged**